src/sdk/namespace_fs.js

```diff
@@ -1127,6 +1134,20 @@ class NamespaceFS {
             throw err;
         }
     }
+
+    async _validate_data_access(fs_account_config, file_path) {
+        try {
+            const stat = await nb_native().fs.stat(fs_account_config, file_path);
+            if (isDirectory(stat)) return;
+            if (stat.blocks * 512 < stat.size) {
```

Comment:
You can't just throw this logic in the code with zero comments.
It's impossible to understand why you are doing this.